include/miopen/miopen.h

```diff
@@ -4704,6 +4706,231 @@ miopenReduceTensor(miopenHandle_t handle,
+ * @param value      Specifies what order should find results have
+ * @return           miopenStatus_t
+ */
+miopenStatus_t miopenSetFindOptionResultsOrder(miopenFindOptions_t options,
+                                               miopenFindResultsOrder_t value);
+
+/*! @brief Sets the workspace limit find option. Default value is maximum of size_t
+ *
+ * @param options    Options object to upfate
+ * @param value      Specifies the workspace limit for find call. All solvers exceeding the limit
+ * would be ignored.
+ * @return           miopenStatus_t
+ */
+miopenStatus_t miopenSetFindOptionWorkspaceLimit(miopenFindOptions_t options, size_t value);
+
+/*! @brief The miopenSolution object describes a prepared solution.
+ */
+MIOPEN_DECLARE_OBJECT(miopenSolution);
+
+/*! @brief Finds solutions to a problem by running different applicable solutions. Memory is
+ * automatically allocated.
+ *
+ * @param handle       Handle to execute the kernels
+ * @param problem      Problem to solve
+ * @param options      Find options. When null default values would be used
+ * @param solutions    Pointer to the first result. Must not be null
+ * @param numSolutions Pointer to the amount of results. Ignored if null
+ * @param maxSolutions Limits the amount of results
+ * @return             miopenStatus_t
+ */
+miopenStatus_t miopenFindSolutions(miopenHandle_t handle,
+                                   miopenProblem_t problem,
+                                   miopenFindOptions_t options,
+                                   miopenSolution_t* solutions,
+                                   size_t* numSolutions,
+                                   size_t maxSolutions);
+
+/*! @brief Values of a tensor argument for the miopenRunSolution function.
+ */
+struct miopenTensorArgument_t
+{
+    /* @brief Identifier of the tensor argument.
+     */
+    miopenTensorArgumentId_t id;
+    /* @brief Tensor descriptor to override the value stored in the solution.
+     *
+     * Some solvers may support overriding input and output tensor descriptors, but right now there
+     * is no way to tell from the API. Intended for the future use.
+     */
+    miopenTensorDescriptor_t* descriptor;
+    /* @brief Pointer to the device memory buffer to use for the operation.
+     */
+    void* buffer;
+};
+
+/*! @brief Runs the solution using the passed in buffers.
+ *
+ * @param handle        Handle to execute the kernels
+ * @param solution      Solution to execute
+ * @param nInputs       Amount to inputs for the solution
+ * @param tensors       Tensor arguments described by miopenTensorArgument_t
+ * @param workspace     Pointer to device buffer used as workspace. May be null when not required.
+ * Should not be less than expected
+ * @param workspaceSize Size of the workspace buffer
+ * @return              miopenStatus_t
+ */
+miopenStatus_t miopenRunSolution(miopenHandle_t handle,
+                                 miopenSolution_t solution,
```

Comment:
`solution`  here, can it be made const ?

Reply:
Why? It's not a reference and unless it won't have any meaningful effect other than a warning. It will become `miopenSolution* const` rather than `const miopenColution*`.

Reply:
ok. 
why would it turn to `miopenSolution* const` instead of `const miopenSolution*` ?

Reply:
Because applying `const` to a type `miopenSolution_t`, which is `miopenSolution*`, would make it `miopenSolution* const`.

Reply:
https://i.imgur.com/Rut5gqz.png

Reply:
For further details you may check literature: https://www.cprogramming.com/tutorial/const_correctness.html

Reply:
`typedef` is not a macro. It defines a type. Application of `const` to a pointer makes the pointer const, not the value.